### PR TITLE
Feature astar implementation

### PR DESCRIPTION
**Pull request by BrandonKroes**
Contains the basis of the entire a* implementation and contains the basis for the memory separation.

### CODE REVIEW

File: module/Structure.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+class GridInFile(Grid):
+
+    def __init__(self, columns=0, rows=0):
+        Grid.__init__(self, columns, rows)
+
+        # The InFile will create it's own file structure on a hard drive (in the module folder)
+        # In Order to prevent collisions we'll be using a random.random AND the epoch time to has a path
+        # While these will prevent MOSTLY prevent collisions, they aren't
+        # immune to it.
+
```
Comment (nicovanbentum) [inline module/Structure.py]: `epoch time to has a path while these will prevent mostly prevent collisions`, what did I just read haha

Comment (BrandonKroes) [reply module/Structure.py]: Should say hash instead has,. but fixed.

File: module/Algorithms.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
+def calculate_heuristic(neighbor, end):
+    """
+    Calculating the heuristic based on coordinates of end and next cell
+    :param neighbor: cell towards the end
+    :param end:  cell where we want to end up
+    :return: int value for the distance
+    """
```
[the remark this answers is not in the document]
Comment (BrandonKroes) [reply module/Algorithms.py]: Are we  using Doxygen for Python code? I can't see it mentioned anywhere on the wiki.
We only have the style guide for code (PEP8) which doesn't have any Docstring requirements, thats what PEP257 is for.

[the remark this answers is not in the document]
Comment (BrandonKroes) [reply module/Algorithms.py]: Implemented.

**Pull request merged**